sympy/matrices/determinant.py

```diff
@@ -538,6 +540,14 @@ def _per(M):
     perm = sympify(perm)
     return perm.simplify()
 
+def _det_DOM(M):
+    DOM = DomainMatrix.from_Matrix(M, field=True, extension=True)
+    K = DOM.domain
+    val = DOM.det()
+    if DOM.domain != EX:
+        val = K.to_sympy(val)
```

Comment:
Why is this checking for the EX domain? Is that necessary? Surely we always want to convert to Expr.

Reply:
Sure it is unnecessary.
Thanks.